src/commit.c

```diff
@@ -65,6 +65,39 @@ Commit_message__get__(Commit *commit)
     return to_unicode(message, encoding, "strict");
 }
 
+PyDoc_STRVAR(Commit_gpg_signature__doc__, "A tuple with the GPG signature and the signed payload.");
+
+PyObject *
+Commit_gpg_signature__get__(Commit *commit)
+{
+    git_buf gpg_signature = { NULL }, signed_data = { NULL };
+    PyObject *py_gpg_signature, *py_signed_data;
+
+    git_oid *oid = git_commit_id(commit->commit);
+    int err = git_commit_extract_signature(
+        &gpg_signature, &signed_data, commit->repo->repo, oid, NULL
+    );
+
+    if (err != GIT_OK){
+        git_buf_free(&gpg_signature);
+        git_buf_free(&signed_data);
+
+        if (err == GIT_ENOTFOUND){
+            Py_INCREF(Py_None); Py_INCREF(Py_None);
+            return Py_BuildValue("OO", Py_None, Py_None);
+        }
+
+        return Error_set(err);
+    }
+
+    py_gpg_signature = PyBytes_FromString(gpg_signature.ptr);
+    py_signed_data = PyBytes_FromString(signed_data.ptr);
+    git_buf_free(&gpg_signature);
+    git_buf_free(&signed_data);
+
+    return Py_BuildValue("OO", py_gpg_signature, py_signed_data);
```

Comment:
Here use **NN** instead of **OO**, because ``PyBytes_FromString`` returns a new reference, otherwise there will be a leak.

Reply:
Yep, didn't realize that `OO` did that (same as `O` vs `N` above. 

Fixed!